server/index.ts

```diff
@@ -0,0 +1,187 @@
+// tslint:disable:no-console
+
+import * as grist from 'grist-plugin-api';
+import {sumBy, times} from 'lodash';
+import * as Shopify from 'shopify-api-node';
+import {applyUpdates, fetchTable, IColumn, ITableData, prepareUpdates} from './dataUpdates';
+
+const storageApi: grist.Storage = grist.rpc.getStub('DocStorage@grist');
+const docApi: grist.GristDocAPI = grist.rpc.getStub('GristDocAPI@grist');
+
+const DestTableId = "ShopifyItems";
+
+// This determines which columns get created in Grist for the destination table.
+const columns = [
+  {id: "ShopifyId",     type: "Text"    },
+  {id: "ItemId",        type: "Text"    },
+  {id: "ProcessedAt",   type: "DateTime"  },
+  {id: "OrderNumber",   type: "Text"      },
+  {id: "VariantTitle",  type: "Text"      },
+  {id: "SKU",           type: "Text"      },
+  {id: "Name",          type: "Text"      },
+  {id: "Title",         type: "Text"      },
+  {id: "Status",        type: "Text"      },
+  {id: "Quantity",      type: "Numeric"   },
+  {id: "Amount",        type: "Numeric"   },
+  {id: "Taxes",         type: "Numeric"   },
+  {id: "Discount",      type: "Numeric"   },
+  {id: "RefundAmount",  type: "Numeric"   },
+  {id: "CreatedAt",     type: "DateTime"  },
+  {id: "UpdatedAt",     type: "DateTime"  },
+  {id: "Date",          type: "Text",   formula: "$CreatedAt.date()" },
+  {id: "Month",         type: "Text",   formula: "$CreatedAt.strftime('%Y-%m')" },
+];
+
+// Type of the record object we extract from the Shopify API.
+interface ILineItem {
+  ShopifyId: string;
+  ItemId: string;
+  ProcessedAt: Date;
+  OrderNumber: string;
+  VariantTitle: string;
+  SKU: string;
+  Name: string;
+  Title: string;
+  Status: string;
+  Quantity: number;
+  Amount: number;
+  Taxes: number;
+  Discount: number;
+  RefundAmount: number;
+  CreatedAt: Date;
+  UpdatedAt: Date;
+}
+
+// Type of the parameters to updateTable() call.
+interface IShopifyParams {
+  startDate: Date;
+  endDate: Date;
+}
+
+// Type of the result of updateTable() call.
+interface IUpdateResult {
+  updated: number;
+  added: number;
+}
+
+function includeTimeZone(type: string, tz: string): string {
+  return (type === 'DateTime' ? `${type}:${tz}` : type);
+}
+
+function includeTimeZones(cols: IColumn[], tz: string): IColumn[] {
+  return cols.map((c) => ({...c, type: includeTimeZone(c.type, tz)}));
+}
+
+// Updates the destination table with data from the Shopify API. Creates the table if needed.
+export async function updateTable(params: IShopifyParams): Promise<IUpdateResult> {
+  try {
+    const credentials = await getCredentials();
+    const shopify = new Shopify(credentials);
```

Comment:
maybe better to set `autoLimit` [option](https://github.com/MONEI/Shopify-api-node#options) to `true` to avoid hitting the [rate limit](https://help.shopify.com/en/api/getting-started/api-call-limit?ref=microapps).

Reply:
No objections and it would be easy to set, but harder to test whether it works. This is unrelated to the refactor, so I'd say better to make a separate commit.